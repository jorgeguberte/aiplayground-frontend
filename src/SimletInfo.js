import React from 'react';
import ReactDOM from 'react-dom';


export default class SimletInfo extends React.Component{
    render(){
        return(
            <div className="SimletInfo">
                <h1 className="SimletInfo-title">{this.props.title}</h1>
                <p className="SimletInfo-description">{this.props.description}</p>
            </div>
        );
    }
}