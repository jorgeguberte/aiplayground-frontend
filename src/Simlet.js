import React from 'react';
import ReactDOM from 'react-dom';

import SimletInfo from './SimletInfo';


export default class Simlet extends React.Component{
    //#INVESTIGATE: How do props trickle down?
    render(){
        return(
            <SimletInfo title={this.props.title} description={this.props.description}/>
        );
    }
}