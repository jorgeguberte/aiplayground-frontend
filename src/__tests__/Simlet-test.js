import React from 'react';
import {shallow, mount, render} from 'enzyme';
import renderer from 'react-test-renderer';


import Simlet from '../Simlet';
import SimletInfo from '../SimletInfo';

/**
 * PROPS TESTING
 */
describe('props integrity', ()=>{
    //Mock props
    const _props = {
        title: 'Simlet Test title', 
        description: 'Simlet Test description'
    };
    const wrapper = mount(
        <Simlet 
            title={_props.title} 
            description={_props.description}
        />
    );
    
    
        describe('title',()=>{
            test('should be defined', ()=>{
                expect(wrapper.props().title).toBeDefined();
            });
            test('should have content', ()=>{
                expect(wrapper.props().title).toBe(_props.title);
            })
        });

        describe('description', ()=>{
            test('should be defined', ()=>{
                expect(wrapper.props().description).toBeDefined();
            });
            test('should have content', ()=>{
                expect(wrapper.props().description).toBe(_props.description);
            })
        });


});



    test('should render', ()=>{
        const component = renderer.create(<Simlet/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });


