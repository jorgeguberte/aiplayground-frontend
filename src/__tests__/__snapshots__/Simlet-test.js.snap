// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="SimletInfo"
>
  <h1
    className="SimletInfo-title"
  />
  <p
    className="SimletInfo-description"
  />
</div>
`;
